spec/lib/aliquot/payment_spec.rb

```diff
@@ -0,0 +1,160 @@
+require 'aliquot-pay'
+
+require 'json'
+require 'openssl'
+
+describe Aliquot::Payment do
+  let(:token) { AliquotPay.generate_token_ecv1(@payment, key, recipient) }
+  let(:token_string) { JSON.unparse(token) }
+  let(:shared_secret) { extract_shared_secret(token, recipient) }
+  let(:cipher) { OpenSSL::Cipher::AES128.new(:CTR) }
+
+  it 'decrypts successfully' do
+    @payment = AliquotPay.payment
+    a = Aliquot::Payment.new(token_string, shared_secret, merchant_id, signing_keys: keystring)
+    a.process
+  end
+
+  it 'rejects expired payment' do
+    @payment = AliquotPay.payment
+    exp_time = (Time.now.to_f - 20).round.to_s
+    @payment[:messageExpiration] = exp_time
+
+    a = Aliquot::Payment.new(token_string, shared_secret, merchant_id, signing_keys: keystring)
+
+    expect { a.process }.to raise_error(Aliquot::TokenExpiredError)
+  end
+
+  it 'rejects invalid token JSON gracefully' do
+    @payment = AliquotPay.payment
+    block = proc do
+      Aliquot::Payment.new('invalid JSON', shared_secret, merchant_id, signing_keys: keystring)
+    end
+
+    expect(&block).to raise_error(Aliquot::InputError, /token JSON invalid, .+/)
```

Comment:
Why not `expect{thing}` rather than `block = proc { thing }; expect(&block)`?

Reply:
Line length :)